### PR TITLE
Create Base Blog

### PR DESCRIPTION
**Pull request by MatiasComercioITBA**
# Summary

- CRUD operations over articles
- CRUD operations over comments, which are nested entities of articles

## Trello Card
https://trello.com/c/chhPgbrN/21-crear-blog-base

### CODE REVIEW

**Comment by epintos** [comment]
@MatiasComercioITBA Be carefull because the git root of this projects its outside the blog folder and it should be inside

File: blog/Gemfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
+source 'https://rubygems.org'
+
```
Comment (epintos) [inline blog/Gemfile]: In a real project, its a good practice to have this file really tidy and with comments

File: blog/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# README
```
Comment (epintos) [inline blog/README.md]: In a real project you will find here lots of things about project bootstrapping, deploy, technologies, tec

File: blog/app/assets/javascripts/articles.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (epintos) [inline blog/app/assets/javascripts/articles.coffee]: The generators you are using generate lots of files you will never use. Its a good practice to erase them. Also a good practice its to make a code review of your own pull request to avoid this like this

File: blog/app/assets/javascripts/comments.coffee [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (epintos) [inline blog/app/assets/javascripts/comments.coffee]: Same

File: blog/app/assets/javascripts/welcome.coffee [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (epintos) [inline blog/app/assets/javascripts/welcome.coffee]: Same

File: blog/app/assets/stylesheets/articles.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Articles controller here.
```
Comment (epintos) [inline blog/app/assets/stylesheets/articles.scss]: Same

File: blog/app/assets/stylesheets/comments.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Comments controller here.
```
Comment (epintos) [inline blog/app/assets/stylesheets/comments.scss]: Same

File: blog/app/assets/stylesheets/welcome.scss [hunk 8]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Welcome controller here.
```
Comment (epintos) [inline blog/app/assets/stylesheets/welcome.scss]: Same

File: blog/app/channels/application_cable/channel.rb [hunk 9]
```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
```
Comment (epintos) [inline blog/app/channels/application_cable/channel.rb]: This files are new in Rails 5 and they are use to make websockets and have notifications in real time for example. But its not part of the training

File: blog/app/controllers/articles_controller.rb [hunk 10]
```diff
@@ -0,0 +1,45 @@
+class ArticlesController < ApplicationController
```
Comment (epintos) [inline blog/app/controllers/articles_controller.rb]: Make sure you have the following sublime configuration for the tabs https://github.com/Wolox/tech-guides/blob/master/sublime-configuration.md

File: blog/app/controllers/articles_controller.rb [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+		@article = Article.new
+	end
+
+	def edit
+		@article = Article.find(params[:id])
+	end
+
+	def create
+		@article = Article.create(article_params)
+		if @article.valid?
```
Comment (epintos) [inline blog/app/controllers/articles_controller.rb]: This was autogenerated? Its the first time i see this approach, usually people use:
```ruby
@article = Article.new(article_params)
if @article.save
.....
```

File: blog/app/controllers/comments_controller.rb [hunk 12]
```diff
@@ -0,0 +1,19 @@
+class CommentsController < ApplicationController
+	def create
```
Comment (epintos) [inline blog/app/controllers/comments_controller.rb]: Pay attention to indentitation. (Not sure if its the Github that is wrong or what)

File: blog/app/controllers/comments_controller.rb [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    @article = Article.find(params[:article_id])
```
Comment (epintos) [inline blog/app/controllers/comments_controller.rb]: See how you have ` @article = Article.find(params[:article_id])` repeated? You can avoid that by using `memorization`

```ruby
def create
   @comment = article.comments.create(comment_params)
    redirect_to article_path(@article)
end

private

def article
   @article ||= Article.find(params[:article_id])
end
```

So it will assign the value to @article the first time when its nil and they it will just use it

File: blog/app/helpers/application_helper.rb [hunk 14]
```diff
@@ -0,0 +1,2 @@
+module ApplicationHelper
```
Comment (epintos) [inline blog/app/helpers/application_helper.rb]: Same

File: blog/app/helpers/comments_helper.rb [hunk 15]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
```
Comment (epintos) [inline blog/app/helpers/comments_helper.rb]: Same

File: blog/app/helpers/articles_helper.rb [hunk 16]
```diff
@@ -0,0 +1,2 @@
+module ArticlesHelper
```
Comment (epintos) [inline blog/app/helpers/articles_helper.rb]: Same

File: blog/app/helpers/welcome_helper.rb [hunk 17]
```diff
@@ -0,0 +1,2 @@
+module WelcomeHelper
```
Comment (epintos) [inline blog/app/helpers/welcome_helper.rb]: Same

File: blog/app/views/articles/_form.html.slim [hunk 18]
```diff
@@ -0,0 +1,23 @@
+= form_for @article do |f|
```
Comment (epintos) [inline blog/app/views/articles/_form.html.slim]: A good practice is to avoid variables with `@` in partials. Take into account that this partial can be used in several places, and a variable with `@` could look like a global variable. You can fix this by calling the partial like this `render 'form', article: @article`

File: blog/db/seeds.rb [hunk 19]
```diff
@@ -0,0 +1,7 @@
+# This file should contain all the record creation needed to seed the database with its default values.
```
Comment (epintos) [inline blog/db/seeds.rb]: Pay attention to this file. Its good to try data. You run it by executing `bundle exec rake db:seed`

File: blog/test/models/article_test.rb [hunk 20]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (epintos) [inline blog/test/models/article_test.rb]: Please remove and add the folder `test` to the gitignore. We are going to use something else later

File: blog/app/controllers/articles_controller.rb [hunk 21]
```diff
@@ -25,20 +25,23 @@ def create
...
 
 	def destroy
-		Article.find(params[:id]).destroy
+		article.destroy
 		redirect_to articles_path
 	end
 
 	private
+		def article
+			@article = Article.find(params[:id])
```
Comment (epintos) [inline blog/app/controllers/articles_controller.rb]: `@article ||=`

File: blog/public/500.html [hunk 22]
```diff
@@ -1,66 +0,0 @@
-<!DOCTYPE html>
```
Comment (epintos) [inline blog/public/500.html]: This file was not necessary. Its usefull when the application fails

Comment (epintos) [inline blog/public/500.html]: `was not necessary to be deleted`

File: blog/public/robots.txt [hunk 23]
```diff
@@ -1,5 +0,0 @@
-# See http://www.robotstxt.org/robotstxt.html for documentation on how to use the robots.txt file
```
Comment (epintos) [inline blog/public/robots.txt]: This is usefull for google robots

File: .gitignore [hunk 24]
```diff
@@ -102,3 +102,7 @@ bower.json
 # Ignore Byebug command history file.
 .byebug_history
 # Rails .gitignore template --
+
+########################################################
+# Own defined rules
+/test/
```
Comment (epintos) [inline .gitignore]: Not sure if it is `test/*`

File: blog/app/channels/application_cable/channel.rb (hunk 9 above)
Comment (MatiasComercioITBA) [inline blog/app/channels/application_cable/channel.rb]: Rails needs this file for rendering OK delete actions

File: blog/app/controllers/articles_controller.rb (hunk 10 above)
Comment (MatiasComercioITBA) [inline blog/app/controllers/articles_controller.rb]: Cannot access this repo... I think it's private.

File: blog/app/controllers/comments_controller.rb (hunk 12 above)
Comment (MatiasComercioITBA) [inline blog/app/controllers/comments_controller.rb]: Github's problem. Code indentation is OK.

**Comment by epintos** [comment]
👍 I will explain you how to squash some commits
